test_scripts/Polices/Policies_Security/Trigger_PTU_NO_Certificate/009_ATF_P_TC_navi_app_PTU_retry_sequence_finish_ForceProtectedService_ON_StartService_encrypted_true.lua

```diff
@@ -0,0 +1,244 @@
+
+  local startserviceEvent = Event()
+  startserviceEvent.matches =
+  function(_, data)
+    return ( data.frameType == 0 and data.serviceType == 11)
+  end
+
+  self.mobileSession:ExpectEvent(startserviceEvent, "Service 11: StartServiceNACK")
+  :ValidIf(function(_, data)
+      if data.frameInfo == 2 then
+        commonFunctions:printError("Service 11: StartServiceACK is received")
+        return false
+      elseif data.frameInfo == 3 then
+        print("Service 11: Audio NACK")
+        return true
+      else
+        commonFunctions:printError("Service 11: StartServiceACK/NACK is not received at all.")
+        return false
+      end
+    end)
+
+  commonTestCases:DelayedExp(10000)
+end
+
+function Test:TestStep_RPC_NACK()
+  self.mobileSession.correlationId = self.mobileSession.correlationId + 1
+
+  local msg = {
+    serviceType = 7,
+    frameInfo = 0,
+    rpcType = 0,
+    rpcFunctionId = 48,
+    encryption = true,
+    rpcCorrelationId = self.mobileSession.correlationId,
+    payload = '{ "audioStreamingIndicator" : "PAUSE" }'
+  }
+  self.mobileSession:Send(msg)
+
+  local startserviceEvent = Event()
+  startserviceEvent.matches =
+  function(_, data)
+    return ( data.frameType == 0 and data.serviceType == 7)
+  end
+
+  self.mobileSession:ExpectEvent(startserviceEvent, "Service 7: StartServiceNACK")
+  :ValidIf(function(_, data)
+      if data.frameInfo == 2 then
+        commonFunctions:printError("Service 7: StartServiceACK is received")
+        return false
+      elseif data.frameInfo == 3 then
+        print("Service 7: Audio NACK")
+        return true
+      else
+        commonFunctions:printError("Service 7: StartServiceACK/NACK is not received at all.")
+        return false
+      end
+    end)
+
+  commonTestCases:DelayedExp(10000)
+end
+
+function Test:TestStep_Hybrid_NACK()
+  self.mobileSession.correlationId = self.mobileSession.correlationId + 1
+
+  local msg = {
+    serviceType = 15,
+    frameInfo = 0,
+    rpcType = 0,
+    rpcFunctionId = 48,
+    encryption = true,
+    rpcCorrelationId = self.mobileSession.correlationId,
+    payload = '{ "audioStreamingIndicator" : "PAUSE" }',
+    binaryData = '{ "audioStreamingIndicator" : "PAUSE" }'
+  }
+  self.mobileSession:Send(msg)
+
+  local startserviceEvent = Event()
+  startserviceEvent.matches =
+  function(_, data)
+    return ( data.frameType == 0 and data.serviceType == 15)
+  end
+
+  self.mobileSession:ExpectEvent(startserviceEvent, "Service 15: StartServiceNACK")
+  :ValidIf(function(_, data)
+      if data.frameInfo == 2 then
+        commonFunctions:printError("Service 15: StartServiceACK is received")
+        return false
+      elseif data.frameInfo == 3 then
+        print("Service 15: Audio NACK")
+        return true
+      else
+        commonFunctions:printError("Service 15: StartServiceACK/NACK is not received at all.")
+        return false
+      end
+    end)
+
+  commonTestCases:DelayedExp(10000)
+end
+
```

Comment:
@istoimenova, It would be better if common code from 4 functions (TestStep_Audio_NACK, TestStep_Video_NACK, TestStep_RPC_NACK, TestStep_Hybrid_NACK) is moved to an additional local function with parameters.